csharp/LogicalBatchReader/ArrayReader.cs

```diff
@@ -140,10 +140,25 @@ public bool HasNext()
             return !_bufferedReader.IsEofDefinition;
         }
 
+        public long Skip(long numRowsToSkip)
+        {
+            for (var i = 0; i < numRowsToSkip; i++)
+            {
+                ReadInnerTypeArray();
```

Comment:
This looks like it should work but for something like an array of integers it'll be pretty inefficient as it reads one item at a time. We should be able to do something slightly better in that case, by using similar logic to `ReadLogicalTypeArray`.

We could possibly optimize this further by supporting it at the `BufferedReader` level and avoiding calling the converter on data we are skipping, but that might get a bit too complicated.

Reply:
Ok, looking into it. Thanks!